plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -823,6 +852,87 @@ public Optional<ConnectorTableLayout> getInsertLayout(ConnectorSession session,
         return Optional.of(new ConnectorTableLayout(partitionColumnNames));
     }
 
+    @Override
+    public Iterator<RelationCommentMetadata> streamRelationComments(ConnectorSession session, Optional<String> schemaName, UnaryOperator<Set<SchemaTableName>> relationFilter)
+    {
+        List<RelationCommentMetadata> views = getViews(session, schemaName).entrySet().stream()
+                .map(entry -> RelationCommentMetadata.forRelation(entry.getKey(), entry.getValue().getComment()))
+                .collect(toImmutableList());
+        Set<SchemaTableName> viewNames = views.stream().map(RelationCommentMetadata::name).collect(toImmutableSet());
+
+        TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+
+        Stream<RelationCommentMetadata> tables = listTables(session, schemaName).stream()
+                .filter(tableName -> !viewNames.contains(tableName))
+                .collect(collectingAndThen(toUnmodifiableSet(), relationFilter)).stream()
+                .map(tableName -> {
+                    if (redirectTable(session, tableName).isPresent()) {
+                        return RelationCommentMetadata.forRedirectedTable(tableName);
+                    }
+
+                    try {
+                        Optional<Table> metastoreTable = metastore.getRawMetastoreTable(tableName.getSchemaName(), tableName.getTableName());
+                        if (metastoreTable.isEmpty()) {
+                            // this may happen when table is being deleted concurrently,
+                            return null;
+                        }
+
+                        Table table = metastoreTable.get();
+                        verifyDeltaLakeTable(table);
+
+                        String tableLocation = HiveMetastoreBackedDeltaLakeMetastore.getTableLocation(table);
+                        if (canUseTableParametersInMetastore(session, fileSystem, table, tableLocation)) {
+                            // Don't check TABLE_COMMENT existence because it's not stored in case of null comment
+                            return RelationCommentMetadata.forRelation(tableName, Optional.ofNullable(table.getParameters().get(TABLE_COMMENT)));
+                        }
+
+                        TableSnapshot snapshot = getSnapshot(session, tableName, tableLocation, Optional.empty());
+                        MetadataEntry metadata = transactionLogAccess.getMetadataEntry(session, snapshot);
+                        return RelationCommentMetadata.forRelation(tableName, Optional.ofNullable(metadata.getDescription()));
+                    }
+                    catch (RuntimeException e) {
+                        boolean silent = false;
```

Comment:
rename to `suppressed` rather than silent. or loggable because that's what happens if this is true